scipy/linalg/tests/test_special_matrices.py

```diff
@@ -640,3 +642,45 @@ def test_fiedler_companion():
     fc = fiedler_companion([1., -16., 86., -176., 105.])
     assert_array_almost_equal(eigvals(fc),
                               np.array([7., 5., 3., 1.]))
+
+
+def test_convolution_matrix():
```

Comment:
normally we'd use a `class` and then place specific tests within that class for pytest namespace resolution

Reply:
It's a singleton so here a class is not needed

Reply:
I don't understand--there are actually several things being tested below--normally one would split each thing being tested into its own specific **unit** test classified within the same test class.

Reply:
I agree with @tylerjereddy, the tests should be split up, and the big loop at the end should use pytest's `parametrize` feature.  The increased granularity lets us know all the things that are going wrong if there is a failure, instead of just the first thing that goes wrong.